app/src/main/java/com/yml/womensafety/adapters/SafetyTipsAdapter.kt

```diff
@@ -3,34 +3,66 @@ package com.yml.womensafety.adapters
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageButton
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.yml.womensafety.R
 import com.yml.womensafety.navigationdrawer.SafetyTips
 
-//Adapter to display Tips for women safety in recycler view
-class SafetyTipsAdapter(private val tipsList: ArrayList<SafetyTips>) :
+class SafetyTipsAdapter() :
     RecyclerView.Adapter<SafetyTipsAdapter.SafetyTipsViewHolder>() {
+    private val tipsList = mutableListOf<SafetyTips>()
 
     class SafetyTipsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        var tips: TextView = itemView.findViewById(R.id.tipsText)
+
+        var title: TextView = itemView.findViewById(R.id.lawTitle)
+        var description: TextView = itemView.findViewById(R.id.lawDescription)
+        var btnExpandDescription: ImageButton =
+            itemView.findViewById(R.id.btnExpandLawDescription)
     }
 
     override fun onCreateViewHolder(
         parent: ViewGroup,
         viewType: Int
     ): SafetyTipsViewHolder {
         return SafetyTipsViewHolder(
-            LayoutInflater.from(parent.context).inflate(R.layout.safety_tip_item, parent, false)
+            LayoutInflater.from(parent.context).inflate(R.layout.law_item, parent, false)
         )
     }
 
-    override fun onBindViewHolder(holder: SafetyTipsViewHolder, position: Int) {
+    override fun onBindViewHolder(
+        holder: SafetyTipsViewHolder,
+        position: Int
+    ) {
         val safetyTips: SafetyTips = tipsList[position]
-        holder.tips.text = holder.tips.context.getString(safetyTips.tipID)
+        var isBtnExpanded: Boolean = tipsList[position].expandable
```

Comment:
Why do we require a global var here? Cant it be local?

Reply:
Made it as local variable